packages/connection/src/class-manager.php

```diff
@@ -759,10 +774,17 @@ public function api_url( $relative_url ) {
 	 * @return String XMLRPC API URL.
 	 */
 	public function xmlrpc_api_url() {
+		$api_base_hook   = 'jetpack_constant_JETPACK__API_BASE';
+		$api_filter_name = __NAMESPACE__ . '\Utils::jetpack_api_constant_filter';
+
+		add_filter( $api_base_hook, $api_filter_name, 10, 2 );
+		$api_base = Constants::get_constant( 'JETPACK__API_BASE' );
+		remove_filter( $api_base_hook, $api_filter_name, 10 );
```

Comment:
Does it seem to you like this implementation adds quite a bit of boilerplate code? Every time we are calling `get_constant`, we are now adding and removing the filter, sometimes twice in the same method. What about using one filter method everywhere? More on that in the other comment.

Reply:
I updated this in 0b39a42. Thanks for the thorough review! Yeah, you're right; there's too much boilerplate code. In the new commit, I added the filter in `Manager::configure()` and in `Client::remote_request()`. It's probably not necessary to remove the filter, since the filter only changes the constant value when the input value is null and the default value exists.